src/skills/Keeper.cpp

```diff
@@ -133,13 +133,25 @@ void Keeper::setGoalPosWithAttacker(RobotPtr attacker) {
     Angle targetAngle = Vector2((i1 + i2)*0.5 - attacker->pos).toAngle();
     end = start + (Vector2) {distanceToGoal*1.2, 0}.rotate(targetAngle);
 
-    auto field = world::field->get_field();
-    Vector2 startGoal = {- field.field_length()/2, - field.goal_width()/2};
-    Vector2 endGoal = {- field.field_length()/2, field.goal_width()/2};
+    auto fieldMsg = field->get_field();
+    Vector2 startGoal = {- fieldMsg.field_length()/2, - fieldMsg.goal_width()/2};
+    Vector2 endGoal = {- fieldMsg.field_length()/2, fieldMsg.goal_width()/2};
     if (control::ControlUtils::lineSegmentsIntersect(start, end, startGoal, endGoal)) {
         goalPos = control::ControlUtils::twoLineIntersection(start, end, startGoal, endGoal);
     }
 }
 
+rtt::Arc Keeper::createKeeperArc() {
```

Comment:
The static reference to world and field is not removed inside this newly added code